lib/client-assets.php

```diff
@@ -487,6 +487,10 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 		'before'
 	);
 
+	// Export data required by the Custom HTML block.
+	wp_add_inline_script( 'wp-editor', sprintf( 'wp.editor.canUnfilteredHtml = %s;', wp_json_encode( current_user_can( 'unfiltered_html' ) ) ) );
```

Comment:
Can we merge this in with other editor settings? I'm thinking we need a scalable solution here.

https://github.com/WordPress/gutenberg/blob/1cea1fbc30658ce2fc135f68503d9ace39bed201/lib/client-assets.php#L790-L793

The `withEditorSettings` higher-order component was created to avoid having the component directly reference the window global:

https://github.com/WordPress/gutenberg/tree/master/blocks/with-editor-settings

Reply:
I absolutely agree. The `withEditorSettings` was not available at the time I wrote this, I believe.